src/components/sidebar/useEditChatTitle.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+import { useUpdateChatTitle } from '@/components/chats/useUpdateChatTitle'
+
+export const useEditChatTitle = (id: string, initialValue: string) => {
+  const [value, setValue] = React.useState<string>(initialValue)
+
+  const { mutate } = useUpdateChatTitle()
+
+  React.useEffect(() => {
+    if (value && id) {
+      void mutate({ id: id, title: value })
+    }
+
+    // Clean on unmount
+    return () => {
+      mutate.cancel()
+    }
+  }, [value, id])
```

Comment:
Not so sure about including the function as a dependency array here.

Reply:
I understand why you would want to ignore it.

The reason to add it here is mainly for code consistency. I think it's better (as in safer, more robust) to respect the "exhaustive-deps" rule everywhere, rather than decide on a case by case basis which dependencies to include and which ones not to.

Some extra context here:
https://legacy.reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Also, [the next PR will throw an eslint error](https://github.com/joiahq/joia/pull/44) if there are any missing dependencies in hooks